pkg/graveler/graveler.go

```diff
@@ -1451,47 +1451,43 @@ func (g *KVGraveler) Delete(ctx context.Context, repository *RepositoryRecord, b
 		return ErrWriteToProtectedBranch
 	}
 
-	deleteEntry := func(branch *Branch) error {
-		commit, err := g.RefManager.GetCommit(ctx, repository, branch.CommitID)
-		if err != nil {
-			return err
-		}
-
-		// check key in committed - do we need tombstone?
-		foundInCommitted := false
-		_, err = g.CommittedManager.Get(ctx, repository.StorageNamespace, commit.MetaRangeID, key)
-		if err != nil {
-			if !errors.Is(err, ErrNotFound) {
-				// unknown error
-				return fmt.Errorf("reading from committed: %w", err)
+	return g.safeBranchWrite(ctx, g.log.WithField("key", key).WithField("operation", "delete"),
+		repository, branchID, func(branch *Branch) error {
+			commit, err := g.RefManager.GetCommit(ctx, repository, branch.CommitID)
+			if err != nil {
+				return err
 			}
-		} else {
-			foundInCommitted = true
-		}
 
-		foundInStaging := true
-		val, err := g.getFromStagingArea(ctx, branch, key)
-		if err != nil {
-			if !errors.Is(err, ErrNotFound) {
-				// unknown error
-				return fmt.Errorf("reading from staging: %w", err)
+			// check staging for entry or tombstone
```

Comment:
Can be performed after the staging read.

Reply:
Done - thanks